packages/cw1155/src/query.rs

```diff
@@ -0,0 +1,41 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cosmwasm_std::{HumanAddr, Uint128};
+
+use crate::msg::TokenId;
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+#[serde(rename_all = "snake_case")]
+pub enum Cw1155QueryMsg {
+    /// Returns the current balance of the given address, 0 if unset.
+    /// Return type: BalanceResponse.
+    Balance { owner: HumanAddr, token_id: TokenId },
+    /// Returns the current balance of the given address for a batch of tokens, 0 if unset.
+    /// Return type: BatchBalanceResponse.
+    BatchBalance {
+        owner: HumanAddr,
+        token_ids: Vec<TokenId>,
+    },
+    /// Queries the approval status of an operator for a given owner
+    /// Return type: ApprovedForAllResponse.
```

Comment:
Again from cw721, how about iterating (page-wise) over all approvals. Here is a template for one way (and you can look at the implementation in cw721-base): https://github.com/CosmWasm/cosmwasm-plus/blob/master/packages/cw721/src/query.rs#L17-L25

Reply:
Looks good, copied those from CW721 spec.